{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "create an experiment CSV file\n",
    "located at /broad/hptmp/jlchang/dogChIPQCtest1/ChIPQC_experiment\n",
    "contents:\n",
    "\"SampleID\",\"Tissue\",\"Factor\",\"Replicate\",\"bamReads\",\"bamControl\",\"ControlID\",\"Peaks\"\n",
    "\"D57_H3K27ac\",\"placenta\",\"H3K27ac\",\"1\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3K27ac_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3_merge.bam\",\"D57_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57chip_out/Peaks/bed/D57_H3K27ac_merge.bed\"\n",
    "\"D57_H3K27me3\",\"placenta\",\"H3K27me3\",\"1\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3K27me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3_merge.bam\",\"D57_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57chip_out/Peaks/bed/D57_H3K27me3_merge.bed\"\n",
    "\"D57_H3k36me3\",\"placenta\",\"H3k36me3\",\"1\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3K36me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3_merge.bam\",\"D57_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57chip_out/Peaks/bed/D57_H3K36me3_merge.bed\"\n",
    "\"D57_H3K4me1\",\"placenta\",\"H3K4me1\",\"1\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3K4me1_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3_merge.bam\",\"D57_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57chip_out/Peaks/bed/D57_H3K4me1_merge.bed\"\n",
    "\"D57_H3K4me3\",\"placenta\",\"H3K4me3\",\"1\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3K4me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57/D57_H3_merge.bam\",\"D57_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D57chip_out/Peaks/bed/D57_H3K4me3_merge.bed\"\n",
    "\"D70_H3K27ac\",\"placenta\",\"H3K27ac\",\"2\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3K27ac_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3_merge.bam\",\"D70_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70chip_out/Peaks/bed/D70_H3K27ac_merge.bed\"\n",
    "\"D70_H3K27me3\",\"placenta\",\"H3K27me3\",\"2\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3K27me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3_merge.bam\",\"D70_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70chip_out/Peaks/bed/D70_H3K27me3_merge.bed\"\n",
    "\"D70_H3k36me3\",\"placenta\",\"H3k36me3\",\"2\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3K36me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3_merge.bam\",\"D70_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70chip_out/Peaks/bed/D70_H3K36me3_merge.bed\"\n",
    "\"D70_H3K4me1\",\"placenta\",\"H3K4me1\",\"2\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3K4me1_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3_merge.bam\",\"D70_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70chip_out/Peaks/bed/D70_H3K4me1_merge.bed\"\n",
    "\"D70_H3K4me3\",\"placenta\",\"H3K4me3\",\"2\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3K4me3_merge.bam\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70/D70_H3_merge.bam\",\"D70_H3\",\"/btl/projects/advancedlibraryprep/chip_seq/dog/C1-83_2016-09-04_2017-02-15/D70chip_out/Peaks/bed/D70_H3K4me3_merge.bed\"\n",
    "\n",
    "Ross' file did not use quotes and was fine\n",
    "\n",
    "use .r-3.3.2\n",
    "R\n",
    "library(ChIPQC)\n",
    "foo <- ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/ChIPQC_experiment.csv\",,\"chr1\")\n",
    "ChIPQCreport(experiment = \"foo\", facet=TRUE,reportName=\"dogChIPQC\",reportFolder=\"ChIPQC\",facetBy=\"Factor\",colourBy=c(\"Replicate\"))\n",
    "\n",
    "\n",
    "\n",
    "Had to delete empty bed files and bed files that only had headers\n",
    "humantestK27me3 <- ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK27me3.csv\",,\"1\")\n",
    "ChIPQCreport(humantestK27me3,reportFolder=\"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK27me3\", facet=FALSE)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "#failed when providing calls file instead of bed file\n",
    "R command line test\n",
    "use .r-3.3.2\n",
    "Rlibrary(ChIPQC)\n",
    "ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/ChIPQC_experiment.csv\",,\"chr1\")\n",
    "#no genome annotation specified (not available for dog)\n",
    "#should ask Ross for repeatmasker blacklist\n",
    "> ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/ChIPQC_experiment.csv\",,\"chr1\")\n",
    "D57_H3K27ac placenta H3K27ac   1 fp4\n",
    "D57_H3K27me3 placenta H3K27me3   1 fp4\n",
    "D57_H3k36me3 placenta H3k36me3   1 fp4\n",
    "D57_H3K4me1 placenta H3K4me1   1 fp4\n",
    "D57_H3K4me3 placenta H3K4me3   1 fp4\n",
    "D70_H3K27ac placenta H3K27ac   1 fp4\n",
    "D70_H3K27me3 placenta H3K27me3   1 fp4\n",
    "D70_H3k36me3 placenta H3k36me3   1 fp4\n",
    "D70_H3K4me1 placenta H3K4me1   1 fp4\n",
    "D70_H3K4me3 placenta H3K4me3   1 fp4\n",
    "Computing metrics for 12 samples...\n",
    "list\n",
    "Bam file has 39 contigs\n",
    "Calculating coverage histogram for chr1\n",
    "\n",
    "Calculating SSD for chr1\n",
    "\n",
    "Calculating unique positions per strand for chr1\n",
    "\n",
    "Calculating shift for chr1\n",
    "\n",
    " 300 / 300\n",
    "Error in names(res) <- nms :\n",
    "  'names' attribute [12] must be the same length as the vector [4]\n",
    "In addition: Warning message:\n",
    "stop worker failed:\n",
    "  'clear_cluster' receive data failed:\n",
    "  reached elapsed time limit\n",
    "> Error in serialize(data, node$con, xdr = FALSE) : ignoring SIGPIPE signal\n",
    "Error in serialize(data, node$con, xdr = FALSE) : ignoring SIGPIPE signal\n",
    "In addition: Warning messages:\n",
    "1: In if (is.na(peaks)) peaks = NULL :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "2: In GetGRanges(bedFile, as.vector(names(ChrLengths)), names(ChrLengths)[k]) :\n",
    "  NAs introduced by coercion\n",
    "Error in serialize(data, node$con, xdr = FALSE) : ignoring SIGPIPE signal\n",
    "In addition: Warning messages:\n",
    "1: In if (is.na(peaks)) peaks = NULL :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "2: In GetGRanges(bedFile, as.vector(names(ChrLengths)), names(ChrLengths)[k]) :\n",
    "  NAs introduced by coercion\n",
    "Error in serialize(data, node$con, xdr = FALSE) : ignoring SIGPIPE signal\n",
    "In addition: Warning messages:\n",
    "1: In if (is.na(peaks)) peaks = NULL :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "2: In GetGRanges(bedFile, as.vector(names(ChrLengths)), names(ChrLengths)[k]) :\n",
    "  NAs introduced by coercion\n",
    "Error in serialize(data, node$con, xdr = FALSE) : ignoring SIGPIPE signal\n",
    "In addition: Warning messages:\n",
    "1: In if (is.na(peaks)) peaks = NULL :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "2: In GetGRanges(bedFile, as.vector(names(ChrLengths)), names(ChrLengths)[k]) :\n",
    "  NAs introduced by coercion"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "signature(experiment = \"ChIPQCexperiment\", facet=TRUE, reportName=\"ChIPQC\", reportFolder=\"ChIPQC\",facetBy=c(\"Tissue\",\"Factor\",\"Condition\",\"Treatment\"),colourBy=c(\"Replicate\"))}}\n",
    "\n",
    "experiment \\tab \\code{\\link{ChIPQCexperiment}} object\\cr\n",
    "facet \\tab logical indicating whether or not to facet using experimental metadata.\\cr\n",
    "reportName \\tab filename of main report file (.html).\\cr\n",
    "reportFolder \\tab directory name where plot graphics will be saved\\cr\n",
    "facetBy \\tab metadata fields to use for faceting\\cr\n",
    "colourBy \\tab metadata field to color by\\cr"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
