{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "20170512\n",
    "use .r-3.3.2\n",
    "R\n",
    "library(ChIPQC)\n",
    "load(\"/broad/hptmp/jlchang/dogChIPQCtest1/humantest2Ab/ChIPQC.RData\")\n",
    "#loads report object, not data object, oh well\n",
    "\n",
    "humantestK4me3 <- ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK4me3.csv\",,\"1\")\n",
    "save(humantestK4me3, file= \"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK4me3/humantestK4me3.RData\")\n",
    "\n",
    "humantestK4me3_all <- ChIPQC(\"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK4me3.csv\")\n",
    "save(humantestK4me3_all, file= \"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK4me3/humantestK4me3_all.RData\")\n",
    "\n",
    "ChIPQCreport(experiment = \"foo\", facet=TRUE,reportName=\"dogChIPQC\",reportFolder=\"ChIPQC\",facetBy=\"Factor\",colourBy=c(\"Replicate\"))\n",
    "\n",
    "\n",
    "\n",
    "Had to delete empty bed files and bed files that only had headers and bed that had low number of peaks\n",
    "\n",
    "ChIPQCreport(humantestK4me3,reportFolder=\"/broad/hptmp/jlchang/dogChIPQCtest1/humantestK4me3\", facet=FALSE)"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "20170516\n",
    "moved dog and humantest data to /cil/shed/sandboxes/jlchang/ChIPseq/\n",
    "unpacked CTCF data from alon's tarball of 2016-04-07-HeLa-CTCF-RNA-Pol-II\n",
    "created bedgraph files from tdfs\n",
    "use igvtools_2.3\n",
    "igvtools tdftobedgraph  file.tdf file.bedgraph\n",
    "converted bedgraph files to bed format\n",
    "awk ' BEGIN { OFS = \"\\t\"; } {  print $1, $2, $3, \"id-\"NR, $4; } ' myData.bedgraph > myData.bed\n",
    "\n",
    "\n",
    "on lager (around 42G at 5th sample; up to 78G during 6th sample; during \"computing metrics\" 231G)\n",
    "use .r-3.3.2\n",
    "R\n",
    "library(ChIPQC)\n",
    "CTCF <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFtest.csv\")\n",
    "Therno_1ul H1 CTCF   1 bed\n",
    "Thermo_3ul H1 CTCF   1 bed\n",
    "Thermo_10ul H1 CTCF   1 bed\n",
    "CST_1ul H1 CTCF   1 bed\n",
    "CST_3ul H1 CTCF   1 bed\n",
    "CST_10ul H1 CTCF   1 bed\n",
    "Checking chromosomes:\n",
    "[1] \"chr1\"\n",
    "Computing metrics for 6 samples...\n",
    "list\n",
    "Bam file has 85 contigs\n",
    "Error in names(res) <- nms :\n",
    "  'names' attribute [6] must be the same length as the vector [5]\n",
    "In addition: Warning message:\n",
    "stop worker failed:\n",
    "  'clear_cluster' receive data failed:\n",
    "  reached elapsed time limit\n",
    "(took ~2.5 hrs to run)\n",
    "\n",
    "(according to hpcstats, peak memory usage ~275G)\n",
    "CTCF_chr1 <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFtest.csv\",\"hg19\",\"1\")\n",
    "ChIPQCreport(CTCF_chr1,reportFolder=\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCF_chr1report\", facet=FALSE)\n",
    "chr1 report generation using 141G so far\n",
    "http://hpcstats.broadinstitute.org/HPC/?c=default&h=lager.broadinstitute.org&m=cpu_report&r=hour&s=descending&hc=4&mc=2\n",
    "\n",
    "c(\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\")\n",
    "\n",
    "c(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"X\",\"Y\")\n",
    "\n",
    "reformat bed file to use chromosome names without \"chr\"\n",
    "\n",
    "reformat hg19 annotation to use chromosome names without \"chr\"\n",
    "annohg19 <- ChIPQC:::getAnnotation(\"hg19\",NULL)\n",
    "for(k in 2:length(annohg19)){seqlevelsStyle(annohg19[[k]]) <- \"NCBI\"}\n",
    "https://support.bioconductor.org/p/65687/\n",
    "\n",
    "CTCF_all <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFreformat.csv\",\"annohg19\",c(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"X\",\"Y\"))\n",
    "Error in getAnnotation(annotation, AllChr = chromosomes) :\n",
    "  Unsupported annotation:annohg19\n",
    "\n",
    "remove annohg19 request\n",
    "CTCF_all <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFreformat.csv\",c(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"X\",\"Y\"))\n",
    "\n",
    "Thermo_1ul 1ul Thermo   1 bed\n",
    "Thermo_3ul 3ul Thermo   1 bed\n",
    "Thermo_10ul 10ul Thermo   1 bed\n",
    "CST_1ul 1ul CST   1 bed\n",
    "CST_3ul 3ul CST   1 bed\n",
    "CST_10ul 10ul CST   1 bed\n",
    "Checking chromosomes:\n",
    "[1] \"1\"\n",
    "Compiling annotation...\n",
    "Error in getAnnotation(annotation, AllChr = chromosomes) :\n",
    "  Unsupported annotation:12345678910111213141516171819202122XY\n",
    "In addition: Warning messages:\n",
    "1: In if (GeneAnnotation == \"hg19\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "2: In if (GeneAnnotation == \"hg38\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "3: In if (GeneAnnotation == \"hg18\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "4: In if (GeneAnnotation == \"mm10\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "5: In if (GeneAnnotation == \"mm9\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "6: In if (GeneAnnotation == \"rn4\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "7: In if (GeneAnnotation == \"ce6\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "8: In if (GeneAnnotation == \"dm3\") { :\n",
    "  the condition has length > 1 and only the first element will be used\n",
    "  \n",
    "specify chromosomes variable\n",
    "CTCF_all <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFreformat.csv\",chromosomes=c(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"X\",\"Y\"))\n",
    "\n",
    "Computing metrics for 6 samples... according to free, using 417G of memory!\n",
    "ran for hours, hit 528G used, errored with:\n",
    "Calculating coverage\n",
    "Calculating Summits on  2  ..\n",
    "Error in names(res) <- nms :\n",
    "  'names' attribute [6] must be the same length as the vector [1]\n",
    "In addition: Warning message:\n",
    "stop worker failed:\n",
    "  'clear_cluster' receive data failed:\n",
    "  reached elapsed time limit\n",
    "  \n",
    "Now that bed names match bam names, try:\n",
    "CTCF_all <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFreformat.csv\")"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "try custom annotation\n",
    "CTCF_all <- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCFreformat.csv\",chromosomes=c(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"X\",\"Y\"))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "save(CTCF_all, file= \"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCF_all.Rdata\")\n",
    "ChIPQCreport(CTCF_chr1,\"annohg19\", reportFolder=\"/cil/shed/sandboxes/jlchang/ChIPseq/CTCF/CTCF_all\", facet=c(\"Factor\",\"Tissue\"))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "after ish (request 10G memory)\n",
    "exp13<- ChIPQC(\"/cil/shed/sandboxes/jlchang/ChIPseq/2016-02-11-exp13-scn/exp13test.csv\")\n",
    "\n",
    "(with default 1G memory)\n",
    "500 Cells CTCF   1 bed\n",
    "Killed\n",
    "\n",
    "(with 10G memory)\n",
    "500 Cells CTCF   1 bed\n",
    "1000 Cells CTCF   1 bed\n",
    "10000 Cells CTCF   1 bed\n",
    "250000 Cells CTCF   1 bed\n",
    "S5 HeLa CTCF   1 bed\n",
    "Killed\n",
    "\n",
    "requested 45G\n",
    "/cil/shed/sandboxes/jlchang/ChIPseq/2016-02-11-exp13-scn 527 $ qsub runR.sh\n",
    "Your job 10181641 (\"runR.sh\") has been submitted\n",
    "Computing metrics for 5 samples...\n",
    "Error in value[[3L]](cond) : setting worker timeout:\n",
    "  error reading from connection\n",
    "Calls: ChIPQC ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>\n",
    "Execution halted\n",
    "\n",
    "remove S5 HeLa CTCF sample and try again\n",
    "Your job 10280601 (\"runR.sh\") has been submitted\n",
    "Error in serialize(data, node$con, xdr = FALSE) :\n",
    "  error writing to connection\n",
    "Calls: ChIPQC ... .send_EXEC -> <Anonymous> -> sendData.SOCK0node -> serialize\n",
    "possibly out of memory issue (or restricted to one core issue?)\n",
    "Job ran for ~20m, maxvmem      83.537G; ru_maxrss    22004796\n",
    "\n",
    "increase memory request to 85G\n",
    "change \"Tissue\" from Cells to the number of cells (for faceting)\n",
    "Your job 10285985 (\"runR.sh\") has been submitted\n",
    "Error in names(res) <- nms :\n",
    "  'names' attribute [4] must be the same length as the vector [3]\n",
    "Calls: ChIPQC -> bplapply -> bplapply -> bplapply -> bplapply\n",
    "In addition: Warning message:\n",
    "stop worker failed:\n",
    "  'clear_cluster' receive data failed:\n",
    "  reached elapsed time limit\n",
    "Execution halted\n",
    "Runtime ~36min;ru_maxrss    22004760, maxvmem      83.537G\n",
    "\n",
    "edit run.R to operate on chromosome names without \"chr\"\n",
    "reformat bed file to use chromosome names without \"chr\"\n",
    "sed 's/chr//g' orig > new\n",
    "fixed the cvs file to run on new bed files (was set to run on bedgraph - possibly why I got the errors)\n",
    "/cil/shed/sandboxes/jlchang/ChIPseq/2016-02-11-exp13-scn 554 $ qsub runR.sh\n",
    "Your job 10304483 (\"runR.sh\") has been submitted - thought chr string was unsupported annotation request (~40m)\n",
    "Your job 10379828 (\"runR.sh\") has been submitted\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "made several corrections to runR.sh and run.R\n",
    "\n",
    "in run.R to prevent BiocParallel from going wide:\n",
    "library(BiocParallel)\n",
    "register(SerialParam())\n",
    "\n",
    "in runR.sh\n",
    "\n",
    " job 10513350 +1_chipseq exited after 14hrs, ru_maxrss    89127380"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
